src/methods/join.js

```diff
@@ -367,6 +367,53 @@ export class JoinImplementation {
     }
   }
 
+  joinPossiblyNormalCompletions(
+    realm: Realm,
+    joinCondition: AbstractValue,
+    c: PossiblyNormalCompletion,
+    a: PossiblyNormalCompletion
+  ): PossiblyNormalCompletion {
+    let rJoinCondition: Value;
+    let cp: [Effects, Effects];
+    let ap: [Effects, Effects];
+    if (c.consequent instanceof AbruptCompletion) {
+      if (a.consequent instanceof AbruptCompletion) {
+        rJoinCondition = AbstractValue.createFromLogicalOp(realm, "&&", c.joinCondition, a.joinCondition);
+        cp = [c.consequentEffects, a.consequentEffects];
+        ap = [c.alternateEffects, a.alternateEffects];
+      } else {
+        let notA = AbstractValue.createFromUnaryOp(realm, "!", a.joinCondition);
+        rJoinCondition = AbstractValue.createFromLogicalOp(realm, "&&", c.joinCondition, notA);
+        cp = [c.consequentEffects, a.alternateEffects];
+        ap = [c.alternateEffects, a.consequentEffects];
+      }
+    } else {
+      let notC = AbstractValue.createFromUnaryOp(realm, "!", c.joinCondition);
+      if (a.consequent instanceof AbruptCompletion) {
+        rJoinCondition = AbstractValue.createFromLogicalOp(realm, "&&", notC, a.joinCondition);
+        cp = [c.alternateEffects, a.consequentEffects];
+        ap = [c.consequentEffects, a.alternateEffects];
+      } else {
+        let notA = AbstractValue.createFromUnaryOp(realm, "!", a.joinCondition);
+        rJoinCondition = AbstractValue.createFromLogicalOp(realm, "&&", notC, notA);
+        cp = [c.alternateEffects, a.alternateEffects];
+        ap = [c.alternateEffects, a.consequentEffects];
```

Comment:
This should be `ap = [c.consequentEffects, ...];` I believe.

Reply:
Ah yes. Thanks for having my back here. I'll fix it and add some more test cases.